swan-lake/integration-tutorials/sending-a-message-to-a-service.md

```diff
@@ -35,63 +35,79 @@ Follow the instructions given in this section to develop the service.
 
-5. Define a record corresponding to the payload from the backend service.
+    ![Paste JSON as record](/learn/images/integration-tutorials/sending-a-message-to-a-service/paste_json_as_record.gif)
 
-    ```ballerina
-    type Doctor record {|
-        string name;
-        string hospital;
-        string category;
-        string availability;
-        decimal fee;
-    |};
-    ```
+5. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and responds to the request.
 
-    The payload will be an array of JSON objects in which, the structure of each JSON object matches this record. Note that you can use the "Paste JSON as record" VS Code command to generate the record if you have the JSON payload.
+    - Open [Ballerina HTTP API Designer](https://wso2.com/ballerina/vscode/docs/design-the-services/http-api-designer) in VS Code
+   
+    - Use `/healthcare` as the service path (or the context) of the service, which is attached to the listener listening on port `port`.
 
-6. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and responds to the request. Use `/healthcare` as the service path (or the context) of the service, which is attached to the listener listening on port `port`. Define an HTTP resource that allows the `GET` operation on resource path `/querydoctor` and accepts the `category` (corresponding to the specialization) as a path parameter.
+        ![Define the service](/learn/images/integration-tutorials/sending-a-message-to-a-service/define_a_service.gif)
 
-    ```ballerina
-    service /healthcare on new http:Listener(port) {
-        resource function get querydoctor/[string category]() {
-            
+    - Define an HTTP resource that allows the `GET` operation on resource path `/doctors` and accepts the `category` path parameter (corresponding to the specialization).
+
+        ![Define the resource](/learn/images/integration-tutorials/sending-a-message-to-a-service/define_a_resource.gif)
+
+        The generated service will be as follows.
+    
+        ```ballerina
+        service /healthcare on new http:Listener(port) {
+            resource function get doctors/[string category]() 
+                    returns Doctor[]|http:NotFound|http:InternalServerError {
+                
+            }
         }
-    }
+        ```
+
+6. Define an [`http:Client`](https://ballerina.io/learn/by-example/#http-client) object to send requests to the backend service.
+
+    ![Define the client](/learn/images/integration-tutorials/sending-a-message-to-a-service/define_client_endpoint.gif)
+
+    The generated code will be as follows.
+
+    ```ballerina
+    final http:Client queryDoctorEP = check new (url = healthcareBackend);
```

Comment:
Let's get rid of the `url = ` part and pass a positional argument here. Let's create a tooling issue for this too.

Reply:
Doing this in GIFs will distract the viewer from the base idea, hence let's add this to complete source code. WDYT?